unittests/AST/ASTImporterTest.cpp

```diff
@@ -4620,6 +4620,63 @@ TEST_P(ASTImporterOptionSpecificTestBase,
   EXPECT_EQ(ToFwdE->getCanonicalDecl(), ToDefE->getCanonicalDecl());
 }
 
+TEST_P(ASTImporterOptionSpecificTestBase, LambdaInFunctionBody) {
+  Decl *FromTU = getTuDecl(
+      R"(
+      void f() {
+        auto L = [](){};
+      }
+      )",
+      Lang_CXX11, "input0.cc");
+  auto Pattern = lambdaExpr();
+  CXXRecordDecl *FromL =
+      FirstDeclMatcher<LambdaExpr>().match(FromTU, Pattern)->getLambdaClass();
+
+  auto ToL = Import(FromL, Lang_CXX11);
+  unsigned ToLSize = 0;
+  unsigned FromLSize = 0;
+  // operator- is not defined for decl_iterator, so we have to traverse and
+  // count.
+  for (auto &D : ToL->decls()) {
+    (void)D;
+    ++ToLSize;
+  }
+  for (auto &D : FromL->decls()) {
+    (void)D;
+    ++FromLSize;
+  }
+  EXPECT_NE(ToLSize, 0u);
+  EXPECT_EQ(ToLSize, FromLSize);
+}
+
+TEST_P(ASTImporterOptionSpecificTestBase, LambdaInFunctionParam) {
+  Decl *FromTU = getTuDecl(
+      R"(
+      template <typename F>
+      void f(F L = [](){}) {}
```

Comment:
Is it possible to use `auto` instead of `F` as parameter type, and omit the template?

Reply:
No that is not possible, auto cannot be part of a function's type:
```
input0.cc:2:14: error: 'auto' not allowed in function prototype
      void f(auto L = [](){}) {}
```